cpp/tests/groupby/groupby_test_util.hpp

```diff
@@ -33,7 +34,8 @@ void test_single_agg(cudf::column_view const& keys,
                      cudf::sorted keys_are_sorted                 = cudf::sorted::NO,
                      std::vector<cudf::order> const& column_order = {},
                      std::vector<cudf::null_order> const& null_precedence = {},
-                     cudf::sorted reference_keys_are_sorted               = cudf::sorted::NO);
+                     cudf::sorted reference_keys_are_sorted               = cudf::sorted::NO,
+                     rmm::cuda_stream_view test_stream = cudf::get_default_stream());
```

Comment:
Note that the other tests in https://github.com/rapidsai/cudf/tree/branch-23.08/cpp/tests/streams are using `cudf::test::get_default_stream()`.

Reply:
Line#166 does use `cudf::test::get_default_stream()`.

That said, I have moved the streams test to a separate file under `tests/streams`. This test uses `cudf::test::get_default_stream()`.

Reply:
Essentially the requirement for streams testing is that in every code path it must be _possible_ to pass `cudf::test::get_default_stream()` through. https://github.com/rapidsai/cudf/pull/13506 contains a number of changes to existing cudf test utilities where there is no input stream so the code had to be changed to always use `cudf::test::get_default_stream`. Since this util is now supporting passing a stream it's fine for this to be `cudf::get_default_stream()`.

That said, it would probably be easier to remove the stream parameter here and just always use `cudf::test::get_default_stream()` internally in this util. [`cudf::test::get_default_stream()` is just an alias for `cudf::get_default_stream()` under normal circumstances](https://github.com/rapidsai/cudf/blob/branch-23.08/cpp/tests/utilities/default_stream.cpp#L24). It exists to provide a symbol that can be overridden to use a different stream when `STREAM_MODE testing` is set in the CMake configuration for a particular test. Using `cudf::test::get_default_stream()` will therefore "just work" in all cases.